File: back/src/main/java/com/back/domain/concert/dto/ConcertListResponse.java
package com.back.domain.concert.dto;

import com.back.domain.concert.entity.Concert;
import java.time.LocalDateTime;

public record ConcertListResponse(
        Long concertId,
        String concertName,
        String venueName,
        LocalDateTime startDate,
        LocalDateTime endDate,
        String imageUrl,
        String status
) {
    public static ConcertListResponse of(Concert concert, String venueName) {
        String status = concert.getEndDate().isAfter(LocalDateTime.now()) ? "AVAILABLE" : "CLOSED";
        return new ConcertListResponse(
                concert.getConcertId(),
                concert.getConcertName(),
                venueName,
                concert.getStartDate(),
                concert.getEndDate(),
                concert.getUrlPoster(),
                status
        );
    }
}

File: back/src/main/kotlin/com/back/domain/concert/dto/ConcertListResponse.kt
package com.back.domain.concert.dto

import com.back.domain.concert.entity.Concert
import java.time.LocalDateTime

data class ConcertListResponse(
    val concertId: Long,
    val concertName: String,
    val venueName: String,
    val startDate: LocalDateTime,
    val endDate: LocalDateTime,
    val imageUrl: String?,
    val status: String
) {
    companion object {
        fun of(concert: Concert, venueName: String): ConcertListResponse = ConcertListResponse(
            concertId = checkNotNull(concert.concertId) { "Concert ID null" },
            concertName = concert.concertName,
            venueName = venueName,
            startDate = concert.startDate,
            endDate = concert.endDate,
            imageUrl = concert.urlPoster,
            status = if (concert.endDate.isAfter(LocalDateTime.now())) "AVAILABLE" else "CLOSED"
        )
    }
}
